lib/rubocop/cop/rspec/implicit_subject.rb

```diff
@@ -66,32 +98,71 @@ def on_send(node)
         private
 
         def autocorrect(corrector, node)
-          replacement = 'expect(subject)'
           case node.method_name
+          when :expect
+            corrector.replace(node, 'is_expected')
+          when :is_expected
+            corrector.replace(node.location.selector, 'expect(subject)')
           when :should
-            replacement += '.to'
+            corrector.replace(node.location.selector, 'expect(subject).to')
           when :should_not
-            replacement += '.not_to'
+            corrector.replace(node.location.selector, 'expect(subject).not_to')
           end
-
-          corrector.replace(node.loc.selector, replacement)
         end
 
-        def valid_usage?(node)
-          example = node.ancestors.find { |parent| example?(parent) }
-          return false if example.nil?
-
-          example.method?(:its) || allowed_by_style?(example)
+        def message(_node)
+          case style
+          when :require_implicit
+            MSG_REQUIRE_IMPLICIT
+          else
+            MSG_REQUIRE_EXPLICIT
+          end
         end
 
-        def allowed_by_style?(example)
+        def invalid?(node)
           case style
+          when :require_implicit
+            invalid_to_require_implicit?(node)
```

Comment:
We need some better naming. E.g. 
invalid_to_require_implicit? -> explicit_unnamed_subject? (directly, without intermediate method, as the name is quite clear)
invalid_to_disallow? -> invalid_implicit_subject?
invalid_to_single_line_only? -> invalid_implicit_subject? && multiline?
invalid_to_single_statement_only? -> invalid_implicit_subject? && multi_statement?

Reply:
if the && creates too much complexity we can move those to methods, but my point is to name them closer to what they are

Reply:
OK, as I said at [another comment](https://github.com/rubocop/rubocop-rspec/pull/1372#discussion_r967707599) we still need the mapping methods to avoid some Metrics cops, so I decided to rename the methods from style-derived names to ones that describe whay they do.